src/h3/mod.rs

```diff
@@ -268,6 +268,8 @@ pub const APPLICATION_PROTOCOL: &[u8] = b"\x05h3-25\x05h3-24\x05h3-23";
 /// [`Result`]: https://doc.rust-lang.org/std/result/enum.Result.html
 pub type Result<T> = std::result::Result<T, Error>;
 
+const MAX_STREAM_GREASE_OVERHEAD: usize = 36;
```

Comment:
AFAICT this is use only in the case of sending GREASE frames, which can contain arbitrarily long data. It's just that the the value 36 is driven only by our choice to send 2 grease frames with one containing some payload of a specific length. Since we only do that once per connection in `send_grease_frames()`, can't we calculate the value there?

Reply:
I realized that we don't actually have to allocate the whole buffer on the stack, as we only need to write one varint at a time, so using a const is not really necessary (otherwise we would have had to allocate the buffer on the heap).

I now changed it to calculate the grease length dynamically, and reduced the size of the buffer.